### ISSUE TITLE
Empty <table> element with PHP8.0 triggers fatal error

### ISSUE THREAD

**Issue by alexmigf** [dompdf/dompdf#2770]
We are facing an issue when an empty `<table>` element is displayed in the PDF using **PHP 8.0** and **DOMPDF** **1.2.0**, triggering a fatal error:

```
Fatal error: Division by zero
```

This doesn't happen using **PHP 7.4**.

The affecting line appears to be this one: https://github.com/dompdf/dompdf/blob/513a6cd872134394f3ffa2292570c038f431026d/src/FrameReflower/Table.php#L169

### Tests

1. This crashes:

```html
<html>
<head>
	<style type="text/css">
		table {
			width: 100%;
			border-spacing: 0;
		}
		td {
			padding: 0;
		}
	</style>
</head>
<body>
	<table></table>
</body>
</html>
```

2. This crashes too:

```html
<html>
<head>
	<style type="text/css">
		table {
			width: 100%;
			border-spacing: 0;
		}
		td {
			padding: 0;
		}
	</style>
</head>
<body>
	<table><tr><td></td></tr></table>
</body>
</html>
```

3. This doesn't crash:

```html
<html>
<head>
</head>
<body>
	<table style="width:100%"></table>
</body>
</html>
```

4. This doesn't crash too:

```html
<html>
<head>
	<style type="text/css">
		table {
			width: 100%;
		}
		td {
			padding: 0;
			margin: 0;
		}
	</style>
</head>
<body>
	<table><tr><td></td></tr></table>
</body>
</html>
```

**Comment by Spreeuw** [comment]
This appears to be a regression of #1057

**Comment by bsweeney** [comment]
I don't know that it's a regression, necessarily. In PHP versions prior to 8.0 division by zero was a warning, not fatal. It just looks like we never addressed the underlying problem because we didn't have a good test case with 1057. Now we do.

FYI the full stack trace is:
```
PHP Fatal error:  Uncaught DivisionByZeroError: Division by zero in /dompdf/dompdf_1-1-1/src/FrameReflower/Table.php:149
Stack trace:
0. /dompdf/dompdf_1-1-1/src/FrameReflower/Table.php(416): Dompdf\FrameReflower\Table->_assign_widths()
1. /dompdf/dompdf_1-1-1/src/FrameDecorator/AbstractFrameDecorator.php(865): Dompdf\FrameReflower\Table->reflow(Object(Dompdf\FrameDecorator\Block))
2. /dompdf/dompdf_1-1-1/src/FrameReflower/Block.php(871): Dompdf\FrameDecorator\AbstractFrameDecorator->reflow(Object(Dompdf\FrameDecorator\Block))
3. /dompdf/dompdf_1-1-1/src/FrameDecorator/AbstractFrameDecorator.php(865): Dompdf\FrameReflower\Block->reflow(NULL)
4. /dompdf/dompdf_1-1-1/src/FrameReflower/Page.php(141): Dompdf\FrameDecorator\AbstractFrameDecorator->reflow()
5. /dompdf/dompdf_1-1-1/src/FrameDecorator/AbstractFrameDecorator.php(865): Dompdf\FrameReflower\Page->reflow(NULL)
6. /dompdf/dompdf_1-1-1/src/Dompdf.php(837): Dompdf\FrameDecorator\AbstractFrameDecorator->reflow()
7. /eclecticgeek.com/dompdf/debug-renderer.php(285): Dompdf\Dompdf->render()
8. {main}
  thrown in /dompdf/dompdf_1-1-1/src/FrameReflower/Table.php on line 149
```

**Comment by Spreeuw** [comment]
> I don't know that it's a regression, necessarily. In PHP versions prior to 8.0 division by zero was a warning, not fatal.

The issue didn't happen in 1.1.1 though (in no PHP version), I thought it was a regression introduced in #2652. Not that it matters much of course (we don't need to get into the semantics), I just wanted to link that issue for visibility.

**Comment by Mellthas** [comment]
> The issue didn't happen in 1.1.1 though (in no PHP version), I thought it was a regression introduced in #2652.

The particular cases posted above are regressions, but with just a slight modification to the first test case, using an additional empty cell, prior versions can be forced to crash as well:

```html
<html>
<head>
	<style type="text/css">
		table {
			width: 100%;
			border-spacing: 0;
		}
		td {
			padding: 0;
		}
	</style>
</head>
<body>
	<table><tr><td></td><td></td></tr></table>
</body>
</html>
```

Anyway, this should be all addressed by #2792.
